cpp/src/groupby/hash/groupby.cu

```diff
@@ -154,43 +249,16 @@ void sparse_to_dense_results(std::vector<aggregation_request> const& requests,
         return std::move(dense_result_table->release()[0]);
       };
 
-    // Enables conversion of ARGMIN/ARGMAX into MIN/MAX
-    auto transformed_result = [&col, to_dense_agg_result, mr, stream](auto const& agg_kind) {
-      auto transformed_agg = std::make_unique<aggregation>(agg_kind);
-      auto arg_result      = to_dense_agg_result(*transformed_agg);
-      // We make a view of ARG(MIN/MAX) result without a null mask and gather
-      // using this map. The values in data buffer of ARG(MIN/MAX) result
-      // corresponding to null values was initialized to ARG(MIN/MAX)_SENTINEL
-      // which is an out of bounds index value (-1) and causes the gathered
-      // value to be null.
-      column_view null_removed_map(
-        data_type(type_to_id<size_type>()),
-        arg_result->size(),
-        static_cast<void const*>(arg_result->view().template data<size_type>()));
-      auto transformed_result =
-        cudf::detail::gather(table_view({col}),
-                             null_removed_map,
-                             arg_result->nullable() ? cudf::detail::out_of_bounds_policy::IGNORE
-                                                    : cudf::detail::out_of_bounds_policy::NULLIFY,
-                             cudf::detail::negative_index_policy::NOT_ALLOWED,
-                             mr,
-                             stream);
-      return std::move(transformed_result->release()[0]);
-    };
+    auto finalizer = hash_compound_agg_finalizer<decltype(to_dense_agg_result)>(
```

Comment:
The `to_desnse_agg_result` makes this logic really difficult to parse because of the lack of explicit dependency injection at the point is is called in `hash_compound_agg_finalizer`. I think this needs to be simplified.

Reply:
Done